skimage/measure/_regionprops.py

```diff
@@ -227,6 +233,19 @@ def euler_number(self):
     def extent(self):
         return self.area / self.image.size
 
+    @property
+    def feret_diameter(self):
+        from ..morphology.convex_hull import convex_hull_image
+        identity_convex_hull = np.pad(self.convex_image,
+                                      2, mode='constant', constant_values=0)
```

Comment:
@emmanuelle is this padding really necessary??? This points to a bug in `find_contours`, no? Or...?

I also wonder whether we should skip convex hull computation in 3D, where it is quite expensive. It might actually be faster to use the coordinates from marching cubes directly.

Reply:
I don't know if it's a bug, but yes `find_contours(np.ones((5, 5)), 0.5)` gives an error (I guess because it's not possible to decide what is outside). I add to add this padding otherwise there can be errors for an image full of ones.

For the 3d case, do you mean retrieving coordinates of the surface from marching cubes, and then compute the convex hull of these points?

Reply:
A region can be a rectangle which results in a `self.convex_image` which is all ones, correct? To prevent problems with `find_contours` in that case, the padding is definitely necessary, I think.

Reply:
@emmanuelle 

> find_contours(np.ones((5, 5)), 0.5) gives an error (I guess because it's not possible to decide what is outside)

I feel like the contour of the image should be the contour then! (ie the implicit padding should be 0 in this case). I'll raise an issue, hopefully we can fix this is the future and get rid of this padding.

> For the 3d case, do you mean retrieving coordinates of the surface from marching cubes, and then compute the convex hull of these points?

I didn't even think that computing the hull would be necessary, as the number of points returned by marching cubes might itself be reasonably small, but yes! =) Ideally, compare the speed of the three methods (convex hull -> marching cubes -> distances, marching cubes -> distances, marching cubes -> convex hull -> distances) for some dummy data (e.g. the binary blobs) and see whether there are any clear winners...

Reply:
I agree with the implicit padding for marching cubes.

For the convex hull, the Feret diameter is defined on the convex hull.

Reply:
@emmanuelle 

> For the convex hull, the Feret diameter is defined on the convex hull.

The feret diameter is defined for any object, the convex hull just makes it faster to compute since the largest diameter must join two vertices of the hull. But the question is that whether "hull + feret on hull vertices" is faster than "feret on all vertices". I guess the former is faster for most data given existing implementations.